api/controllers/recipe.js

```diff
@@ -2,140 +2,121 @@ const Recipe = require("../models/recipe");
+  } catch (error) {
+    res.statusCode(500).json(error);
+  }
 };
-exports.getUpdateViews = (req, res, next) => {
-  Recipe.findOneAndUpdate({ _id: res._id }, { $inc: { views: 1 } })
-    .then((recipe) => {
-      res.status(200).json(recipe);
-    })
-    .catch((err) => {
-      console.log(err);
-    });
+
+exports.getSearchRecipesByTitle = async (req, res, next) => {
+  try {
+    const searchQuery = req.query.q.trim();
+    if (searchQuery.length <= 1) {
+      return res.status(403).json("Query must be at least 2 characters");
+    }
+    let { sort } = req.query;
+    const recipes = await Recipe.find({
+      title: { $regex: searchQuery },
+    }).sort({ [sort]: -1 });
+    res.status(200).json({ recipes, total: recipes.length });
+  } catch (error) {
+    res.statusCode(500).json(error);
+  }
 };
 
-exports.createRecipe = (req, res, next) => {
-  const {
-    title,
-    user,
-    imgUrl,
-    photographer,
-    ingredients,
-    instructions,
-    prepTime,
-    servings,
-    description,
-    sourceName,
-    sourceUrl,
-    difficulty,
-    categories,
-  } = req.body;
-  let creator;
-  const recipe = new Recipe({
-    title,
-    creator: "61832f8dbe0844faaf3cab41",
-    // creator: req.userId,
-    imgUrl,
-    photographer,
-    ingredients,
-    instructions,
-    prepTime,
-    servings,
-    description,
-    sourceName,
-    sourceUrl,
-    difficulty,
-    categories,
-  });
-  recipe
-    .save()
-    .then((result) => User.findById("61832f8dbe0844faaf3cab41"))
-    // .then((result) => User.findById(req.userId))
-    .then((user) => {
-      creator = user;
-      user.recipes.push(recipe);
-      return user.save();
-    })
-    .then((result) => {
-      res
-        .status(201)
-        .json({
-          message: "Recipe created successfully",
-          recipe: recipe,
-          creator: { _id: creator._id, name: creator.name },
-        });
-    })
-    .catch((err) => {
-      console.log(err);
+exports.getSingleRecipe = async (req, res, next) => {
+  try {
+    const recipe = await Recipe.findOneAndUpdate(
+      { _id: req.params.recipeId },
+      { $inc: { views: 1 } }
+    );
+    res.status(200).json(recipe);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+};
+
+exports.createRecipe = async (req, res, next) => {
+  try {
+    const {
+      title,
+      userId,
+      imgUrl,
+      photographer,
+      ingredients,
+      instructions,
+      prepTime,
+      servings,
+      description,
+      sourceName,
+      sourceUrl,
+      categories,
+    } = req.body;
+    const recipe = await new Recipe({
+      title,
+      creator: userId,
+      imgUrl,
+      photographer,
+      ingredients,
+      instructions,
+      prepTime,
+      servings,
+      description,
+      sourceName,
+      sourceUrl,
+      categories,
+    });
+    await recipe.save();
+    const user = await User.findById(userId);
+    user.recipes.push(recipe);
+    await user.save();
```

Comment:
There's a difficult subject called transactions which handle the issue of a recipe being created but not added to user so it's removing the recipe but it's not for now I guess